app/src/main/java/openfoodfacts/github/scrachx/openfood/features/HomeFragment.kt

```diff
@@ -197,25 +199,25 @@ class HomeFragment : NavigationBaseFragment() {
                 .subscribeOn(Schedulers.io()) // io for network
                 .observeOn(AndroidSchedulers.mainThread()) // Move to main thread for UI changes
                 .doOnError { Log.e(LOG_TAG, "Could not retrieve tag-line from server.", it) }
-                .subscribe { languages ->
-                    val localAsString = LocaleHelper.getLocaleFromContext(context).toString()
+                .subscribe { tagLines ->
+                    val appLanguage = localeManager.getLanguage()
                     var isLanguageFound = false
                     var isExactLanguageFound = false
-                    languages.forEach { tagLine ->
-                        val languageCountry = tagLine.language
-                        if (!isExactLanguageFound && (languageCountry == localAsString || languageCountry.contains(localAsString))) {
-                            isExactLanguageFound = languageCountry == localAsString
-                            taglineURL = tagLine.tagLine.url
-                            binding.tvDailyFoodFact.text = tagLine.tagLine.message
+                    tagLines.forEach { tag ->
+                        if (!isExactLanguageFound && (tag.language == appLanguage || tag.language.contains(appLanguage))) {
+                            isExactLanguageFound = tag.language == appLanguage
+                            taglineURL = tag.tagLine.url
+                            binding.tvDailyFoodFact.text = tag.tagLine.message
                             isLanguageFound = true
                         }
                     }
                     if (!isLanguageFound) {
-                        taglineURL = languages.last().tagLine.url
-                        binding.tvDailyFoodFact.text = languages.last().tagLine.message
+                        taglineURL = tagLines.last().tagLine.url
+                        binding.tvDailyFoodFact.text = tagLines.last().tagLine.message
```

Comment:
I've never understood the tvDailyFoodFact var name. Can we name it tagline as well, or something else ?

Reply:
sure

Reply:
@naivekook it's a rant from me, it's perfectly fine otherwise ;-)